Added Quart server + Vite/React frontend for testing Sieve functions

anniebabannie opened the pull request:
I cleaned up my work into 3 commits, which I hope makes things easier to review. One commit for the basic requirements in the root, one for the Vite frontend, and one for the server (which I borrowed somewhat from @lazeratops filler word demo!). I did my best to remove some straggler code from the filler-word-removal repo, but please let me know if I missed anything.

Pretty much all of the server code is piled into `server/index.py`, but probably deserves a bit of refactoring. Let me know what you think @lazeratops !

anniebabannie committed:
added README and basic py requirements

anniebabannie committed:
added frontend

anniebabannie committed:
added quart server

lazeratops commented on README.md:
```diff
@@ -0,0 +1,86 @@
+
+1. Clone this repository.
+1. Copy the `.env.sample` file into `.env`. DO NOT submit your `.env` file to version control.
+1. Update your `.env` with your `DAILY_API_KEY`
```
Just to confirm, is deepgram API key also required here? I see it in .sample.env above

anniebabannie replied:
Correct – I'll remove it from the `.sample.env`, thank you!

lazeratops commented on README.md:
```diff
@@ -0,0 +1,86 @@
+
+## How it works
+
+This demo starts by allowing users to fetch their latest Daily video recordings, using the Daily API. Once obtained, users are presented with three choices for running Sieve functions to run on their Daily recordings. This includes the following:
```
```suggestion
This demo starts by enabling users to fetch their latest Daily video recordings, using Daily's REST API. Once obtained, users are presented with three choices for running Sieve functions to run on their Daily recordings. This includes the following:
```

"Allow" is usually more of a permission thing whereas "enable" is more of a "We're giving you the capability to do this" thing.

lazeratops commented on README.md:
```diff
@@ -0,0 +1,86 @@
+
+This demo starts by allowing users to fetch their latest Daily video recordings, using the Daily API. Once obtained, users are presented with three choices for running Sieve functions to run on their Daily recordings. This includes the following:
+
+1. `audio_enhancement` - https://www.sievedata.com/functions/sieve/audio_enhancement
```
Suggest just make the function name itself the link (vs pasting the URL after the name).

Also, this list seems to be reflecting different things. The first two items are referring directly to Sieve functions whereas the third is referring to the overall feature. It might be nice to standardize it like this:

```
1. Audio enhancement using Sieve's [`audio_enhancement`](LINK) function
1. Video lip-sync using Sieve's [`text_to_video_lipsync`](LINK) function
1. Video dubbing, which comprises of _four different Sieve functions_:
  1. [`speech_transcriber`](LINK)
  1. [`seamles_text2text`](LINK)
```

etc... wdyt?

anniebabannie replied:
Works for me!

lazeratops commented on README.md:
```diff
@@ -0,0 +1,86 @@
+  1. `xtts-v1` - https://www.sievedata.com/functions/sieve/xtts-v1
+  1. `video_retalking` - https://www.sievedata.com/functions/sieve/video_retalking
+
+All Sieve functions follow more or less the same usage, which is as follows:
```
```suggestion
All Sieve functions follow more or less the same usage pattern, which is as follows:
```

lazeratops commented on README.md:
```diff
@@ -0,0 +1,86 @@
+
+Closer inspection of this demo code will reveal that not all Daily recordings are passed in to Sieve functions _as is_. 
+
+For example, for both of the lip-syncing demos, we had to shave off the first second of the Daily recording since they begin with a brief black screen instead of the speaker's face. Additionally, the lip-syncing functions tend not to work on videos where there are more than one active speaker. **Input requirements for each Sieve function is clearly outlined in their README files available on Siev ...
```
```suggestion
For example, for both of the lip-syncing demos, we had to shave off the first second of the Daily recording since they begin with a brief black screen instead of the speaker's face. Additionally, the lip-syncing functions tend not to work on videos where there is more than one active speaker. **Input requirements for each Sieve function is clearly outlined in their README files available on Sieve's website**, so be sure to reference those when trouble-shooting.
```

lazeratops commented on server/config.py:
```diff
@@ -0,0 +1,61 @@
+
+UPLOAD_DIR_ENV = 'UPLOAD_DIR'
+OUTPUT_DIR_ENV = 'OUTPUT_DIR'
+TMP_DIR_ENV = 'TEMP_DIR'
```
Are all of these being used? If so, `TMP_DIR` should be added to `.sample.env` as well (and if not we can refer references to it here)

anniebabannie replied:
👍

lazeratops commented on README.md:
```diff
@@ -0,0 +1,86 @@
+
+## Sieve AI Processing of Daily Video Call Recordings
```
Reads very nicely! We usually try to include a screenshot of the demo in the README to give people an idea of what they're about to see. Can we add one?

anniebabannie replied:
Added!

lazeratops commented on server/config.py:
```diff
@@ -0,0 +1,61 @@
+        os.makedirs(directory)
+
+
+def get_upload_dir_path() -> str:
```
If this only works on Daily cloud recordings it might be more intuitive to rename this to `get_recordings_dir_path()` (along with the associated env variable and other "upload" references).

anniebabannie replied:
Good idea! Renamed.

lazeratops commented on server/config.py:
```diff
@@ -0,0 +1,61 @@
+    return os.path.join(get_output_dir_path(), file_name)
+
+
+def get_temp_dir_path() -> str:
```
At a quick ctrl+f I don't see this being used, which is making me think maybe we can remove all the temp dir related stuff in here

anniebabannie replied:
👍

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    return process_text_to_video_lipsync(recording_id, file_path)
+
+@app.route('/audio_enhance/<recording_id>', methods=['POST'])
+async def process_recording_audio_enhance(recording_id):
```
PEP8 generally requires two blank lines between functions in modules and one between methods in classes. I think you already have an autopep8 dependency here in the requirements file, suggest just giving that a run as it'll fix up things like spacing issues, line lengths, etc!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    audio_clip = AudioFileClip(audio_path)
+    return video_clip.set_audio(audio_clip)
+
+def get_new_video_path(vid_id: str, file_type: bool):
```
I think the type hint here isn't correct - the file type I see being passed in is a string. 

```suggestion
def get_new_video_path(vid_id: str, file_type: str) -> str:
```

might be more relevant to rename this to `file_extension` also

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    except Exception as e:
+        return process_error('failed to download Daily recording', e)
+
+    file_path = get_new_video_path(recording_id, 'mp4')
```
Lines 67 to 77 here seem to be repeated across multiple handlers - might be worth just pulling them out into their own function that return a file path.

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+
+        video = sieve.Video(path=video_path)
+        audio = extract_audio(video)
+        print(audio)
```
Do you think this print line is useful? If so, can we provide some more context before printing the audio to make the log line clearer? If not let's remove

anniebabannie replied:
Nah, just some left over debugging prints :)

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        print(audio)
+        output = audio_enhance(audio)
+        
+        json_dict = {
```
Suggest using a more relevant name here, as right now the name is basically the type of object and does not represent what this is meant to be _for_ . Maybe `response`? Since down below you just set `response` to `json_dict` anyway? This could get rid of that final assignment as well.

anniebabannie replied:
👍

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        
+        json_dict = json_dict | { 'processed_video': f'{recording_id}-enhanced.mp4', 'original_video': original_name}
+
+        response = json_dict
```
Yeah, if you rename `json_dict` to response when you first create it up above you could just remove this line.

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+            recording_id: recording_id
+        }
+        
+        print(output.path)
```
Suggest adding a little bit of context to this print statement. E.g.: `print(f"enhanced audio output path: {output.path}")` (or something)

lazeratops replied:
Does not need to block this PR, I am just going to unresolved this for my own reference in case it makes sense to go back and pick up the straggling comments next week :)

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        video_clip = VideoFileClip(video_path)
+        audio_clip = AudioFileClip(output.path)
+        final_clip = video_clip.set_audio(audio_clip)
+        new_video_path = f'./vite/public/{recording_id}-enhanced.mp4'
```
I have not used vite much, from what I understand it is a frontend framework? If so, is it relevant to hard-code front-end related paths here, _or_ is it more suitable to use the environment variables we already process in the configuration up above to decide on where to store videos (e.g.., `OUTPUT_DIR` as seen in the sample env file)? ATM the user seems to find themselves unable to configure where things are stored on their file system like this.

anniebabannie replied:
Setting an OUTPUT_DIR is definitely a better idea, I'll do that!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        final_clip.write_videofile(new_video_path)
+        
+        video_name = f'{recording_id}.mp4'
+        original_name = f'original-{video_name}'
```
So we keep the original in order for the user to be play both side by side, is that right?

anniebabannie replied:
Correct!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        
+        video_name = f'{recording_id}.mp4'
+        original_name = f'original-{video_name}'
+        original_path = f'./vite/public/{original_name}'
```
Similar thing here - will defer to your knowledge of vite, but this seems to somewhat negate the benefits of configurable upload paths etc in the environment and possibly blurs the separation between server and client.

anniebabannie replied:
👍

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        clip = VideoFileClip(video_path)
+        duration = clip.duration
+        clip = clip.subclip(1,duration)
+        original_name = f'original-{video_name}'
```
I noticed this `original-` prefix being used multiple times - suggest pulling out the building of this name to a separate function that can be reused, to reduce maintenance and potential for error.

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        duration = clip.duration
+        clip = clip.subclip(1,duration)
+        original_name = f'original-{video_name}'
+        original_trimmed_video = f'./vite/public/{original_name}'
```
Similar question here about vite, but also _if_ this ends up being a legitimate path that is relevant to hard-code, we should pull out the creation of this path into its own function for consistent reuse vs hard-coding it several times. (Also may be safer to make this an absolute path?)

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    lipsyncer = sieve.function.get("sieve/video_retalking:de84f40")
+
+    print("transcribing audio")
+    # transcribe audio
```
```suggestion
```

With the print line above I think this comment doesn't add much

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    print("translated text:", translated_text)
+
+    print("generating tts audio")
+    # Generate new audio from translated text
```
```suggestion
```

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    system_packages=["ffmpeg"],
+)
+
+def video_dubbing(source_video: sieve.Video, language: str):
```
Not a blocker, but in general it would be nice to include return type hints through these funcs as well

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        clip.write_videofile(original_trimmed_video)
+
+        video = sieve.Video(path=original_trimmed_video)
+        path = video_dubbing(video, "spanish")
```
To help the reader follow along it may be nice to name this something slightly more descriptive than `path`.  - `dubbed_path` maybe?

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+
+    print("starting lipsync")
+    # Combine audio and video with Retalker
+    return lipsyncer.run(source_video, target_audio).path
```
Is the output path here something we have set, or something hard-coded by Sieve's module? (i.e., wondering if it is relevant to have this go straight to the desired output location vs doing a move operation in the caller)

anniebabannie replied:
Running `.path` is how you actually download the video from Sieve, and I think I prefer to keep this decoupled from what we actually _do_ with the video path, hope that makes sense?

lazeratops replied:
Ohhh interesting so they have it implemented as a property but it actually performs a download operation? (I wish they had made it a function in that case for clarity but that's nothing you can do anything about on your end!)

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    """Runs filler-word-removal processing on given file."""
+    try:
+        video = sieve.Video(path=video_path)
+        video_transcript_analyzer = sieve.function.get("sieve/video_transcript_analyzer")
```
should probably version-lock this

anniebabannie replied:
Actually I dont even use this function, just forgot to remove it lol!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        video = sieve.Video(path=video_path)
+        video_transcript_analyzer = sieve.function.get("sieve/video_transcript_analyzer")
+        output = video_transcript_analyzer.run(video)
+        json_dict = {
```
Same thought here with renaming this to something more relevant and descriptive. Like before, I would suggest `response` (and removing the setting of a new variable called `response` to this down below)

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        json_path = os.path.join(get_upload_dir_path(), f'{recording_id}.json')
+
+        for i, output_object in enumerate(output):
+            print(i)
```
These prints seem like they are going to get very spammy. Suggest removing this one. If `output_object` and `type(output_object)` are useful to print, suggest putting them on one log line with a descriptive prefix.

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    try:
+        video = sieve.Video(path=video_path)
+        video_transcript_analyzer = sieve.function.get("sieve/video_transcript_analyzer")
+        output = video_transcript_analyzer.run(video)
```
Can we name `output` something  more descriptive? The name does not mean much on it own, which might be fine if we weren't doing anything fancy with it and using it in close succession to the call, but I can see we're doing some very interesting operations with this below and the name can get confusing when trying to understand what is happening. 

Also, "output" here can also be mistaken for what we intend to be the output of the route as a whole. Or a final output file. Or some other kind of output. 

Also maybe add a comment with some guidance on what we expect the output to be, since I see we iterate over it below and I'm not quite sure what it is we're iterating over while reading, so it could be handy to explain in brief.

anniebabannie replied:
I'm actually okay with output being a generic term since every Sieve function's output object is a little different (their docs tend to always use `output` as the variable name too, so I think keeping it this name will make it easier for devs when cross-referencing Sieve's documentation).

lazeratops replied:
Fair enough!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+            print(i)
+            print(output_object)
+            print(type(output_object))
+            if i == 2 or i == 3 or i == 4:
```
I think this part could at least use a comment to explain what we're doing and why here! Why those specific objects?  Why do we ignore 0 and 1? What happens if there are more than 5 elements? 

(Can also write this as follows to simplify: `if 2 <= i <= 4:` )

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        json_dict = {
+            recording_id: recording_id
+        }
+        json_path = os.path.join(get_upload_dir_path(), f'{recording_id}.json')
```
Can we make this variable name slightly more descriptive? It is clear that it will be a json path, but a json path of _what_/for what purpose?

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+                json_dict = json_dict | output_object
+
+        with open(json_path, 'w') as fp:
+            json.dump(json_dict, fp)
```
What is this going to be used for? Being in the uploads dir makes it look like it is not presented back to the user, right?

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    except Exception as e:
+        return process_error('failed to process file – check logs for details', e)
+
+def transcript_to_text(transcript):
```
Is the transcript a certain type of object? Suggest adding type hint (and return type hint, which would be `-> str`

anniebabannie replied:
I don't actually know what type of object this is (again, this is borrowed from Sieve's docs), so I'm going to leave as is ¯\_(ツ)_/¯ Agreed that having a type would be ideal tho

lazeratops replied:
Going to unresolved this not as an action point for you but a reminder for myself to look into this in any potential followup!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+    return text
+
+@sieve.function(
+    name="video-dubber",
```
We've used a few sieve functions up until this point, but this is the first time I see this decorator and it is for use one func. Wondering what it is used for and why just this one?  Worth adding a comment imo

anniebabannie replied:
This is taken from Sieve's documentation, TBH I'm not sure why they wrote it this way? Given that it's how they wrote things, I'm going to leave it for now.

lazeratops replied:
Hmm ok; I'm a little worried that this does not seem to be version locked at all and the demo could break as soon as that function changes? (I think this might be the last real followup to this PR that would be nice to confirm today before merging btw, everything else looks great so far!)

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        raise Exception("Language not supported")
+
+    # use remote sieve models
+    transcriber = sieve.function.get("sieve/speech_transcriber:8eb4cdf")
```
Yay for version locking!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+
+@app.after_serving
+async def shutdown():
+    for task in app.background_tasks:
```
I do not see any background tasks actually being added, so this may not be relevant

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+
+def extract_audio(source_video: sieve.Video):
+    import subprocess
+    audio_path = 'temp.wav'
```
Won't this break by overwriting the same file over and over if someone initiates two processing requests at the same time? Should probably use a unique file name

anniebabannie replied:
This was more borrowed code from Sieve's docs, but yes, it will get overwritten, however as the app I built only displays one video transformation at a time, I don't think it's much of a problem if it gets overwritten.

lazeratops replied:
Ah ok, and it only lets people _start_ one transformation at a time as well?

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+
+@app.route('/projects/<project_id>', methods=['GET'])
+async def get_status(project_id):
+    """Route to return current processing status of a project."""
```
I do not actually see a project status being updated anywhere, nor does it seem like the concept of a project ID exists in this demo? Did I miss it or should this be removed?

anniebabannie replied:
Removing it!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+
+
+@app.route('/projects/<project_id>/download', methods=['GET'])
+async def download_final_output(project_id):
```
Same here - I don't see us writing anything to the provided output dir or returning a project ID to the user. I _do suggest_ using the configured output dir to store processing output vs a hard-coded vite path up above. But if you opt not to go this route and this actually is completely unused, I suggest removing all mentions of an output dir as well as this route (very possible I may be missing something somewhere though!)

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        text_to_video_lipsync = sieve.function.get("sieve/text_to_video_lipsync:58dfd4eb")
+        output = text_to_video_lipsync.run(video, text, tts_model, speech_stability, speech_similarity_boost, elevenlabs_voice_id, elevenlabs_cleanup_voice_id, refine_source_audio, refine_target_audio)
+        
+        print(output)
```
Suggest removing these or making them more descriptive

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        print(output.path)
+        processed_name = f'text-to-video-lipsynced-{video_name}'
+        processed_path = f'./vite/public/{processed_name}'
+        shutil.move(output.path, processed_path)
```
Same questions here as above with the output path being hardcoded to what _seems_ to be a client-specific location? Should this just use the configured output directory instead?

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        clip.write_videofile(original_trimmed_video)
+
+        video = sieve.Video(path=original_trimmed_video)
+        text = "Hello, my name is Annie. I am a human being and definitely not a robot incognito. My hobbies are normal human things like eating bread and playing sports ball. I have no plans to take over the world."
```
LOL :D 

Was this text actually meant to be provided by the user?

anniebabannie replied:
I'm happy leaving this hardcoded text in, I figured if users want to edit the code they can :)

I feel like I'm setting myself up for a meme of dozens of AI videos claiming to be "Annie, who is definitely a human being" lol

lazeratops replied:
Fair enough - it might make sense to call this out in the README and point to the place where users can change it

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        original_trimmed_video = f'./vite/public/original-{video_name}'
+        clip.write_videofile(original_trimmed_video)
+
+        video = sieve.Video(path=original_trimmed_video)
```
I think some of these processing functions could use some guiding comments throughout them.

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        tts_model = "elevenlabs"
+        speech_stability = 0.5
+        speech_similarity_boost = 0.63
+        elevenlabs_voice_id = "21m00Tcm4TlvDq8ikWAM" # ElevenLabs's "Rachel" voice
```
Does Sieve actually handle elevenlabs API access or how does this work? Does the user need to provide an elevenlabs API key somewhere?

anniebabannie replied:
Oh good call actually! I need to make a note that you have to have a Sieve account and add your Elevenlabs secret on their platform (not as a typical env var). Good catch!

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+
+        original_name = f'original-{video_name}'
+        original_path = f'./vite/public/{original_name}'
+        shutil.move(video_path, original_path)
```
Wondering if it may be relevant to get rid of some of these moves by writing straight to the relevant folder.

anniebabannie replied:
I'm not actually sure how, since it's part of Sieve's codebase that when you call `.path` on the output, it automatically downloads to particular path. ¯\_(ツ)_/¯ So for now I'm going to leave it as is, but I agree that it would be better to save it to a path of the user's choosing.

lazeratops commented on server/index.py:
```diff
@@ -0,0 +1,333 @@
+        original_path = f'./vite/public/{original_name}'
+        shutil.move(video_path, original_path)
+
+        json_dict = {
```
Same suggestion here just naming this `response` and getting rid of `response=json_dict` below

lazeratops commented on vite/README.md:
```diff
@@ -0,0 +1,27 @@
+# React + TypeScript + Vite
+
+This template provides a minimal setup to get React working in Vite with HMR and some ESLint rules.
```
Suggest removing this README as it is not relevant to this demo

lazeratops commented on vite/index.html:
```diff
@@ -0,0 +1,13 @@
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Let's make this something more relevant to the demo

lazeratops commented on vite/package.json:
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "vite",
```
Let's make this name more relevant to the demo

lazeratops commented on vite/src/App.tsx:
```diff
@@ -0,0 +1,112 @@
+const apiURL = 'http://127.0.0.1:5000';
+
+
+type DailyRecording = {
```
I love the smell of TypeScript in the morning :D

lazeratops commented on vite/src/App.tsx:
```diff
@@ -0,0 +1,112 @@
+        }
+      })
+      .catch((e) => {
+        console.error('Failed to process uploaded video:', e);
```
This is just fetching recordings, not processing any video right? Let's make this error log more relevant.

lazeratops commented on vite/src/App.tsx:
```diff
@@ -0,0 +1,112 @@
+      setProcessedVideo(data);
+    })
+    .catch((e) => {
+      console.error('Failed to process uploaded video:', e);
```
This is not an uploaded video but a recording - let's make this error log more relevant

lazeratops commented on vite/src/App.tsx:
```diff
@@ -0,0 +1,112 @@
+  }
+
+  function processRecording(endpoint: string, recording_id: string) {
+    fetch(`${apiURL}/${endpoint}/${recording_id}`, { method: 'POST' })
```
I am wondering what the response times are like here from the server, since at first glance at the code (have not run it yet!) it seems like we are doing everything synchronously?  Will the user here have some kind of feedback indicating a pending operation/progress of some sort?

anniebabannie replied:
Ahhh, you're totally right, I was being lazy and forgot to add some "Processing...." UX feedback. Just added that now, nothing fancy but it should do the trick...

lazeratops commented on vite/src/App.tsx:
```diff
@@ -0,0 +1,112 @@
+    fetch(`${apiURL}/${endpoint}/${recording_id}`, { method: 'POST' })
+    .then((res) => {
+      if (res.ok === false) {
+        throw Error(`upload request failed: ${res.status}`);
```
This isn't an upload request, so suggest rewording

lazeratops commented on vite/src/App.tsx:
```diff
@@ -0,0 +1,112 @@
+    fetch(`${apiURL}/recordings`, { method: 'GET' })
+      .then((res) => {
+        if (res.ok === false) {
+          throw Error(`upload request failed: ${res.status}`);
```
This isn't an upload request, suggest rewording

lazeratops commented on vite/src/App.tsx:
```diff
@@ -0,0 +1,112 @@
+      return res.json();
+    })
+    .then((data) => {
+      console.log(data)
```
Suggest adding a descriptive prefix to this log line for clarity

lazeratops commented on vite/tailwind.config.js:
```diff
@@ -0,0 +1,27 @@
+/** @type {import('tailwindcss').Config} */
```
So nice to have all the client-side separated into its own subdir!!

lazeratops reviewed (commented):
Hello! Thank you so much for building the out! I've left some comments here. I have _not_ yet had a chance to run the demo so these are based on reading the code only - please LMK if anything does not make sense or if I've misunderstood any of the components!

anniebabannie committed:
Update README.md

Co-authored-by: Liza <65890040+lazeratops@users.noreply.github.com>

anniebabannie committed:
updated sample env + readme links

anniebabannie committed:
Update README.md

Co-authored-by: Liza <65890040+lazeratops@users.noreply.github.com>

anniebabannie committed:
Update README.md

Co-authored-by: Liza <65890040+lazeratops@users.noreply.github.com>

anniebabannie committed:
cleaned up unused methods from old repo

anniebabannie committed:
Added refactored code from Lizas comments

anniebabannie committed:
PEP8 cleanup

anniebabannie committed:
added screenshot

anniebabannie commented:
@lazeratops I got through addressing most of your comments, there's a couple of stragglers but I need to start filming + editing the demo video, so I hope the remaining comments aren't complete blockers. If you think they are please let me know! Thank you so much for the thorough review!

lazeratops reviewed (approved):
Ty for addressing all the comments! I marked a couple as unresolved just as TODOs for myself to maybe follow up on in another PR next week. The one thing that may be worth looking at still before merging is that decorator-specified sieve function that does not appear to be version locked. Just to make sure there is no chance of it changing under us and breaking the whole demo. (but approving preemptively)

Also, this is based purely from looking at the code itself! I have not run the code at this time and assume you have tested it out after these changes. If not I think it is definitely worth a final sanity test!

anniebabannie committed:
bumped version of sieve function
